house/tests/my_house_add_expense_tests.py

```diff
@@ -0,0 +1,111 @@
+EXPENSE_BAD_FORM_DATA_AMOUNT = {'date': DATE, 'amount': BAD_AMOUNT, 'category': CATEGORY, 'description': DESCRIPTION}
+EXPENSE_BAD_FORM_DATA_DESCRIPTION = {'date': DATE, 'amount': AMOUNT, 'category': CATEGORY,
+                                     'description': BAD_DESCRIPTION}
+
+
+@pytest.fixture
+def request_factory():
+    return RequestFactory()
+
+
+@pytest.fixture
+def generate_get_request(request_factory):
+    get_request = request_factory.get('<str:house_id>/addExpense/')
+    return get_request
+
+
+@pytest.fixture
+def generate_expense_form(request_factory):
+    post_request = request_factory.post('<str:house_id>/addExpense/', EXPENSE_FORM_DATA)
+    form = ExpenseForm(post_request.POST)
+    return form
+
+
+@pytest.fixture
+def generate_bad_expense_form_description(request_factory):
+    post_request = request_factory.post('<str:house_id>/addExpense/', EXPENSE_BAD_FORM_DATA_DESCRIPTION)
+    form = ExpenseForm(post_request.POST)
+    return form
+
+
+@pytest.fixture
+def generate_bad_expense_form_amount(request_factory):
+    post_request = request_factory.post('<str:house_id>/addExpense/', EXPENSE_BAD_FORM_DATA_AMOUNT)
+    form = ExpenseForm(post_request.POST)
+    return form
+
+
+@pytest.mark.django_db
+class TestExpenseForm:
+    def test_expense_form(self, generate_expense_form):
+        form = generate_expense_form
+        assert form.is_valid()
+
+    def test_bad_form_amount(self, generate_bad_expense_form_amount):
+        form = generate_bad_expense_form_amount
+        assert not form.is_valid()
+
+    def test_bad_form_description(self, generate_bad_expense_form_description):
+        form = generate_bad_expense_form_description
+        assert not form.is_valid()
+
+    def test_add_expenses_to_my_house(self, request_factory):
+        house = HouseFactory()
+        house.save()
+        post_request = request_factory.post(f'{house.house_id}/addExpense/', EXPENSE_FORM_DATA)
+        add_expense(post_request, house.house_id)
+        assert len(Expenses.objects.all()) == 1
+
+
+@pytest.mark.django_db
+class TestMyHouseAddExpenseViews:
+    def test_add_expense_function_200(self, generate_get_request):
+        HouseFactory().save()
+        house = House.objects.all()[0]
+        try:
+            render = add_expense(generate_get_request, house.house_id)
+            assert render.status_code == 200, "Status code is not 200"
+        except Exception:
+            assert False, "Error in house_view function"
+
+    def test_mine_page_add_expense_views_templates(self):
+        try:
+            get_template(MINE_ADD_EXPENSE_ROUTE)
+        except TemplateDoesNotExist:
+            assert False, f"Template {MINE_ADD_EXPENSE_ROUTE} does not exist"
+
+    def test_mine_page_add_expense_btns_views_templates(self):
+        try:
+            get_template(MINE_ADD_EXPENSE_BTN_ROUTE)
+        except TemplateDoesNotExist:
+            assert False, f"Template {MINE_ADD_EXPENSE_BTN_ROUTE} does not exist"
```

Comment:
Barak suggested using the client in the test to check the rendering. 
Can you add some tests using it or change the ones you wrote? 
for example: 
https://github.com/redhat-beyond/Xpense/issues/98

Reply:
Following yesterday's meeting, the responsibility will be Yoav's
@Yoavsc129